runzi/configuration/subduction_inversions.py

```diff
@@ -38,111 +38,228 @@ def build_subduction_tasks(general_task_id, rupture_sets, args):
-                slip_rate_weighting_type=slip_rate_weighting_type,
-                slip_rate_normalized_weight=slip_rate_normalized_weight,
-                slip_rate_unnormalized_weight=slip_rate_unnormalized_weight,
-                mfd_mag_gt_5=mfd_mag_gt_5,
-                mfd_b_value=mfd_b_value,
-                mfd_transition_mag=mfd_transition_mag,
-
-                mfd_uncertainty_weight=mfd_uncertainty_weight,
-                mfd_uncertainty_power=mfd_uncertainty_power,
-                mfd_uncertainty_scalar=mfd_uncertainty_scalar,
-
-                #New config arguments for Simulated Annealing ...
-                selection_interval_secs=selection_interval_secs,
-                threads_per_selector=threads_per_selector,
-                averaging_threads=averaging_threads,
-                averaging_interval_secs=averaging_interval_secs,
-                non_negativity_function=non_negativity_function,
-                perturbation_function=perturbation_function,
-
-                scaling_relationship=scaling_relationship,
-                scaling_recalc_mag=scaling_recalc_mag,
-                deformation_model=deformation_model,
-
-                scaling_c_val=scaling_c_val,
-                initial_solution_id=initial_solution_id
-                )
-
-            job_arguments = dict(
-                task_id = task_count,
-                # round = round,
-                java_threads = int(threads_per_selector) * int(averaging_threads),
-                jvm_heap_max = JVM_HEAP_MAX,
-                java_gateway_port=task_factory.get_next_port(),
-                working_path=str(WORK_PATH),
-                root_folder=OPENSHA_ROOT,
-                general_task_id=general_task_id,
-                use_api = USE_API
-                )
-
-            if CLUSTER_MODE == EnvMode['AWS']:
-
-                job_name = f"Runzi-automation-subduction_inversions-{task_count}"
-                config_data = dict(task_arguments=task_arguments, job_arguments=job_arguments)
-
-                yield get_ecs_job_config(job_name, rupture_set_info['id'], config_data,
-                    toshi_api_url=API_URL, toshi_s3_url=S3_URL, toshi_report_bucket=S3_REPORT_BUCKET,
-                    task_module=inversion_solution_builder_task.__name__,
-                    time_minutes=int(max_inversion_time), memory=30720, vcpu=4)
-
-            else:
-                #write a config
-                task_factory.write_task_config(task_arguments, job_arguments)
-
-                script = task_factory.get_task_script()
-
-                script_file_path = PurePath(WORK_PATH, f"task_{task_count}.sh")
-                with open(script_file_path, 'w') as f:
-                    f.write(script)
-
-                #make file executable
-                st = os.stat(script_file_path)
-                os.chmod(script_file_path, st.st_mode | stat.S_IEXEC)
-
-                yield str(script_file_path)
-                #return
+    if args.get('b_and_n'):
```

Comment:
@chrisdicaprio go with this for now, but it's getting a bit unwieldy. Maybe split this up next time around?